packages/admin/admin-rte/src/core/translation/xml/transformStateToXml.ts

```diff
@@ -0,0 +1,171 @@
+import { CharacterMetadata, ContentBlock, ContentState } from "draft-js";
+import type { List } from "immutable";
+
+/*
+    This is a first, very basic implementation, inspired by https://github.com/sstur/draft-js-utils/blob/master/packages/draft-js-export-html/src/stateToHTML.js.
+    Images, code snippets, custom options as attributes, elements or styles are not taken into account.
+*/
+import getEntityRanges from "./getEntityRanges";
+
+type CharacterMetaList = List<CharacterMetadata>;
+
+export const INLINE_STYLE = {
+    BOLD: "BOLD",
+    ITALIC: "ITALIC",
+    STRIKETHROUGH: "STRIKETHROUGH",
+    UNDERLINE: "UNDERLINE",
+    SUB: "SUB",
+    SUP: "SUP",
+};
+
+export const ENTITY_TYPE = {
+    LINK: "LINK",
+};
+
+// Order: inner-most style to outer-most.
+// Examle: <em><strong>foo</strong></em>
+const DEFAULT_STYLE_ORDER = [
+    INLINE_STYLE.BOLD,
+    INLINE_STYLE.ITALIC,
+    INLINE_STYLE.UNDERLINE,
+    INLINE_STYLE.STRIKETHROUGH,
+    INLINE_STYLE.SUB,
+    INLINE_STYLE.SUP,
+];
+
+const DEFAULT_STYLE_MAP = {
+    [INLINE_STYLE.BOLD]: { element: "inline" },
+    [INLINE_STYLE.ITALIC]: { element: "inline" },
+    [INLINE_STYLE.STRIKETHROUGH]: { element: "inline" },
+    [INLINE_STYLE.UNDERLINE]: { element: "inline" },
+    [INLINE_STYLE.SUB]: { element: "inline" },
+    [INLINE_STYLE.SUP]: { element: "inline" },
```

Comment:
Is this flexible enough?

It's possible to add custom inline styles in the project afaik (@johnnyomair ?). So we would have to also support them.

Reply:
see https://github.com/vivid-planet/comet/pull/1587/files#r1455283058